### PR TITLE
User story 19

### LINKED ISSUE #15: User Story 19, User can delete a book
```
As a Visitor,
When I visit a book's show page,
I see a link on the page to delete the book.
This link should return me to the book index page where I
no longer see this book listed.

(your controller may need to delete other content before you can delete the book)
```

DO THESE STORIES FIRST:
- User Story 10, Book Show Page

### Checklist:

- [x] base code tests are written
- [x] base code is written to pass tests
- [x] edge case ("sad path") tests written
- [x] edge case code written
- [x] code is reviewed
- [x] branch is merged
- [x] all tests still pass

child of #9

child of #33

depends on #20

### PR DESCRIPTION
**Pull request by chakeresa**
This PR/branch...
 - Resolves #15 (User Story 19 - destroy a book)
 - Adds button/route/action to destroy a book
 - Automatically destroys the book's children (book_authors & reviews) when the book is destroyed
 - Tests all of the above at the model & feature levels
 - Fixes a couple random typos in other parts of the app

### CODE REVIEW

File: spec/features/books/destroy_spec.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
+require 'rails_helper'
+
+RSpec.describe "As a visitor", type: :feature do
+  describe "when I click the delete button on an book's show page" do
+    before(:each) do
+      @book_1 = Book.create!(title: "The Frozen Deep", page_count: 106, year_published: 1874, thumbnail: "https://images.gr-assets.com/books/1328728986l/1009218.jpg")
+      @book_2 = Book.create!(title: "To Kill a Mockingbird", page_count: 281, year_published: 1960, thumbnail: "https://upload.wikimedia.org/wikipedia/en/7/79/To_Kill_a_Mockingbird.JPG")
+      @author_1 = @book_1.authors.create!(name: "Wilkie Collins")
```
Comment (DavisC0801) [inline spec/features/books/destroy_spec.rb]: These author Ivars are not used and can be cut.

File: spec/models/book_spec.rb [hunk 2]
```diff
@@ -19,6 +19,47 @@
     it {should have_many :reviews}
   end
 
+  describe "class methods" do
+    describe "when destroying a book" do
+      before(:each) do
+        @book_1 = Book.create!(title: "The Frozen Deep", page_count: 106, year_published: 1874, thumbnail: "https://images.gr-assets.com/books/1328728986l/1009218.jpg")
+        @book_2 = Book.create!(title: "To Kill a Mockingbird", page_count: 281, year_published: 1960, thumbnail: "https://upload.wikimedia.org/wikipedia/en/7/79/To_Kill_a_Mockingbird.JPG")
+        @author_1 = @book_1.authors.create!(name: "Wilkie Collins")
```
Comment (DavisC0801) [inline spec/models/book_spec.rb:27]: As above, authors Ivar's not used in this test.

**Review by DavisC0801 — COMMENTED** [review]
Overall, no issues or anything to look at, thank you for refactoring and changing test names for additional clarity.

File: spec/features/books/destroy_spec.rb (hunk 1 above)
Comment (chakeresa) [reply spec/features/books/destroy_spec.rb]: Done -- thanks for catching that

File: spec/models/book_spec.rb (hunk 2 above)
Comment (chakeresa) [reply spec/models/book_spec.rb:27]: Actually these ones are used, because I do assertions on BookAuthor.
```ruby
all_book_ids_in_book_authors = BookAuthor.pluck(:book_id)

expect(all_book_ids_in_book_authors).to_not include(@book_1.id)
expect(all_book_ids_in_book_authors).to include(@book_2.id)
```
